sympy/sets/sets.py

```diff
@@ -165,7 +307,16 @@ def _union(self, other):
 
         Used within the :class:`Union` class
         """
-        return None
+        try:
+            return self._union_simplify(other)
+        except:
+            # There may be the case when Imageset expr have more than one
```

Comment:
@Shekharrajak Adding a comment doesn't [fix the issues with bare except clauses](https://github.com/sympy/sympy/pull/11188/files#r68429752). People find the specific error which is raised for the condition you are describing and specifically catch that.